### PR TITLE
[mpc] add mpc/1.1.0

### PR DESCRIPTION
**Pull request by SSE4**
based on : https://github.com/bincrafters/conan-mpc/tree/stable/1.1.0

Specify library name and version:  **mpc/1.1.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/mpc/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+                args.extend(["--disable-static", "--enable-shared"])
+            else:
+                args.extend(["--disable-shared", "--disable-static"])
+            env_build = AutoToolsBuildEnvironment(self)
+            env_build.configure(args=args)
+            env_build.make(args=["V=0"])
+            env_build.install()
+
+    def package(self):
+        self.copy(pattern="COPYING.LESSER", dst="licenses", src=self._source_subfolder)
```
Comment (madebr) [inline recipes/mpc/all/conanfile.py:49]:
```suggestion
        os.rename(extracted_dir, self._source_subfolder)

    def _configure_autotools(self):
        if self._autotools:
            return self._autotools
        self._autotools = AutoToolsBuildEnvironment(self)
        args = []
        if self.options.shared:
            args.extend(["--disable-static", "--enable-shared"])
        else:
            args.extend(["--disable-shared", "--disable-static"])
        self._autotools.configure(args=args)
        return self._autotools

    def build(self):
        with tools.chdir(self._source_subfolder):
            autotools = self._configure_autotools()
            autotools.make()

    def package(self):
        self.copy(pattern="COPYING.LESSER", dst="licenses", src=self._source_subfolder)
        with tools.chdir(self._source_subfolder):
            autotools = self._configure_autotools()
            autotools.install()
```

File: recipes/mpc/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    homepage = "http://www.multiprecision.org/mpc/home.html"
+    license = "LGPL-3.0-or-later"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+    requires = "mpfr/4.0.2"
+
+    def config_options(self):
```
Comment (madebr) [inline recipes/mpc/all/conanfile.py:21]:
```suggestion
    requires = "mpfr/4.0.2"

    @property
    def _source_subfolder(self):
        return "source_subfolder"

    _autotools = None

    def config_options(self):
```

File: recipes/mpc/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def configure(self):
+        if self.settings.compiler == 'Visual Studio':
```
Comment (madebr) [inline recipes/mpc/all/conanfile.py:26]:
```suggestion
        if self.settings.compiler == "Visual Studio":
```

File: recipes/mpc/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+            env_build.configure(args=args)
+            env_build.make(args=["V=0"])
+            env_build.install()
+
+    def package(self):
+        self.copy(pattern="COPYING.LESSER", dst="licenses", src=self._source_subfolder)
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+        la = os.path.join(self.package_folder, "lib", "libmpc.la")
+        if os.path.isfile(la):
+            os.unlink(la)
```
Comment (madebr) [inline recipes/mpc/all/conanfile.py:53]:
```suggestion
        os.unlink(os.path.join(self.package_folder, "lib", "libmpc.la"))
```

File: recipes/mpc/all/conanfile.py (hunk 3 above)
Comment (SSE4) [reply recipes/mpc/all/conanfile.py:26]: is there any advantage?

**Comment by SSE4** [comment]
@Croydon that's reasonable. I'll keep only GL here, removing GLU and GLES traces.

File: recipes/mpc/all/conanfile.py (hunk 3 above)
Comment (madebr) [reply recipes/mpc/all/conanfile.py:30]: No, this was the only usage of single quotes. And it was sticking in my eyes :smile:

**Comment by Croydon** [comment]
Ew... according to GitHub notifications I was mentioned here, but I can't see any pinging here. So.. did someone want my attention here? 😄

File: recipes/mpc/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+    def _configure_autotools(self):
+        if self._autotools:
+            return self._autotools
+        self._autotools = AutoToolsBuildEnvironment(self)
+        args = []
+        if self.options.shared:
+            args.extend(["--disable-static", "--enable-shared"])
+        else:
+            args.extend(["--disable-shared", "--disable-static"])
```
Comment (jgsogo) [inline recipes/mpc/all/conanfile.py:48]:
```suggestion
            args.extend(["--disable-shared", "--enable-static"])
```

typo?

**Comment by SSE4** [comment]
@Croydon nothing here, I just put comment to the wrong browser tab. sorry for the noise.

File: recipes/mpc/all/test_package/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,10 @@
+cmake_minimum_required(VERSION 2.8.12)
+project(test_package)
+
+set(CMAKE_VERBOSE_MAKEFILE TRUE)
```
Comment (uilianries) [inline recipes/mpc/all/test_package/CMakeLists.txt:4]:
```suggestion
```

File: recipes/mpc/all/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    def _configure_autotools(self):
+        if self._autotools:
+            return self._autotools
+        self._autotools = AutoToolsBuildEnvironment(self)
+        args = []
+        if self.options.shared:
+            args.extend(["--disable-static", "--enable-shared"])
+        else:
+            args.extend(["--disable-shared", "--enable-static"])
+        self._autotools.configure(args=args)
```
Comment (uilianries) [inline recipes/mpc/all/conanfile.py:49]:
```suggestion
        self._autotools.configure(args=args, configure_dir=self._source_subfolder)
```

File: recipes/mpc/all/conanfile.py [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+        return self._autotools
+
+    def build(self):
+        with tools.chdir(self._source_subfolder):
+            autotools = self._configure_autotools()
+            autotools.make()
```
Comment (uilianries) [inline recipes/mpc/all/conanfile.py:55]:
```suggestion
        autotools = self._configure_autotools()
        autotools.make()
```

File: recipes/mpc/all/conanfile.py [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+
+    def package(self):
+        self.copy(pattern="COPYING.LESSER", dst="licenses", src=self._source_subfolder)
+        with tools.chdir(self._source_subfolder):
+            autotools = self._configure_autotools()
+            autotools.install()
```
Comment (uilianries) [inline recipes/mpc/all/conanfile.py:61]:
```suggestion
        autotools = self._configure_autotools()
        autotools.install()
```

**Pull request merged**